src/processes.jl

```diff
@@ -358,11 +364,19 @@ julia> foldl(Filter(isodd), 0.0, 1:4) do state, input
 4.0
 ```
 """
-Base.foldl(step, xform::Transducer, init, itr) =
-    mapfoldl(xform, Completing(step), init, itr)
+function Base.foldl(step, xform::Transducer, itr;
+                    init=mapfoldl_init(xform, step, itr))
+    mapfoldl(xform, Completing(step), itr, init=init)
+end
 
-Base.foldl(step, ed::Eduction, init) =
-    foldl(step, Transducer(ed), init, ed.coll)
+function Base.foldl(step, ed::Eduction;
+                    init=let
+                        xf = ed.rf.xform
```

Comment:
I think this actually should be `xf = Transducer(ed)` (I guess it's a bit tricky) since `ed.rf.xform` is just one "head" transducer.  Instead, we need to re-construct the chain of transducers again.  I know it is redundant but it can be fixed later.

ATM, to get the API surface right, why not just do:

```julia
Base.foldl(step, ed::Eduction; kwargs...) =
    foldl(step, Transducer(ed), ed.coll; kwargs...)
```

(Not sure how to do a multi-line suggestion in browser. Probably I need to edit locally to do that.)